src/euler_types.jl

```diff
@@ -0,0 +1,216 @@
+"""
+type EulerYXZ <: TaitByranOrder end
+
+
+"""
+EulerXYZ corresponds to:
+
+	Extrinsic: R = Rx(theta_x) * Ry(theta_y) * Rz(theta_z)
+
+or equivalently,
+
+	Intrinsic: R = Rz''(theta_z) * Ry'(theta_y) * Rx(theta_x) (where the axis of rotation inherits from previous rotations)
+"""
+type EulerXYZ <: TaitByranOrder end
+
+
+"""
+EulerXZY corresponds to:
+
+	Extrinsic: R = Rx(theta_x) * Rz(theta_z) * Ry(theta_y)
+
+or equivalently,
+
+	Intrinsic: R = Ry''(theta_y) * Rz'(theta_z) * Rx(theta_x) (where the axis of rotation inherits from previous rotations)
+"""
+type EulerXZY <: TaitByranOrder end
+
+
+#
+# proper-euler ordering
+#
+abstract ProperEulerOrder   <: EulerOrder
+
+
+"""
+EulerZYZ corresponds to:
+
+	Extrinsic: R = Rz(theta_1) * Ry(theta_2) * Rz(theta_3)
+
+or equivalently,
+
+	Intrinsic: R = Rz''(theta_3) * Ry'(theta_2) * Rz(theta_1) (where the axis of rotation inherits from previous rotations)
+"""
+type EulerZYZ <: ProperEulerOrder end
+
+
+"""
+EulerZXZ corresponds to:
+
+	Extrinsic: R = Rz(theta_1) * Rx(theta_2) * Rz(theta_3)
+
+or equivalently,
+
+	Intrinsic: R = Rz''(theta_3) * Rx'(theta_2) * Rz(theta_1) (where the axis of rotation inherits from previous rotations)
+"""
+type EulerZXZ <: ProperEulerOrder end
+
+
+"""
+EulerYZY corresponds to:
+
+	Extrinsic: R = Ry(theta_1) * Rz(theta_2) * Ry(theta_3)
+
+or equivalently,
+
+	Intrinsic: R = Ry''(theta_3) * Rz'(theta_2) * Ry(theta_1) (where the axis of rotation inherits from previous rotations)
+"""
+type EulerYZY <: ProperEulerOrder end
+
+
+"""
+EulerYXY corresponds to:
+
+	Extrinsic: R = Ry(theta_1) * Rx(theta_2) * Ry(theta_3)
+
+or equivalently,
+
+	Intrinsic: R = Ry''(theta_3) * Rx'(theta_2) * Ry(theta_1) (where the axis of rotation inherits from previous rotations)
+"""
+type EulerYXY <: ProperEulerOrder end
+
+
+"""
+EulerXYX corresponds to:
+
+	Extrinsic: R = Rx(theta_1) * Ry(theta_2) * Rx(theta_3)
+
+or equivalently,
+
+	Intrinsic: R = Rx''(theta_3) * Ry'(theta_2) * Rx(theta_1) (where the axis of rotation inherits from previous rotations)
+"""
+type EulerXYX <: ProperEulerOrder end
+
+
+"""
+EulerXZX corresponds to:
+
+	Extrinsic: R = Rx(theta_1) * Rz(theta_2) * Rx(theta_3)
+
+or equivalently,
+
+	Intrinsic: R = Rx''(theta_3) * Rz'(theta_2) * Rx(theta_1) (where the axis of rotation inherits from previous rotations)
+"""
+type EulerXZX <: ProperEulerOrder end
+
+
+##############################
+# Euler (Tait Byran) Angles
+##############################
+
+macro DefaultEulerOrder(); EulerZXY; end
+
+"""
+EulerAngles{Order <: TaitByranOrder, T <: AbstractFloat} where order specifies the extrinsic (axes of rotation are fixed) order of rotations.
+See subtypes(Rotations.TaitByranOrder) for a list of possible orders
+
+Fields:
+	theta_x   - Angle to rotate about the x axis (radians)
+	theta_y   - Angle to rotate about the y axis (radians)
+	theta_z   - Angle to rotate about the z axis (radians)
+"""
+immutable EulerAngles{Order <: TaitByranOrder, T <: Real} 
```

Comment:
Is it worth having an abstract type for `EulerAngles` and `ProperEulerAngles`?

Going the other way, why have both? Why not simply use one type parameterized by an `EulerOrder` with a specialization on `ProperEulerOrder` for specific methods only as necessary? It could simplify code in many places, and remove one type from the package.